iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -0,0 +1,402 @@
+                            $" maxTemp={report.maxTemp}, minTemp={report.minTemp}, avgTemp={report.avgTemp}, startTime={report.startTime.LocalDateTime}, " +
+                            $"endTime={report.endTime.LocalDateTime}");
+
+                        byte[] responsePayload = Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(report));
+                        return await Task.FromResult(new MethodResponse(responsePayload, (int)StatusCode.Completed));
+                    }
+
+                    _logger.LogDebug($"Command: component=\"{componentName}\", no relevant readings found since {sinceInDateTimeOffset.LocalDateTime}, " +
+                        $"cannot generate any report.");
+                    return await Task.FromResult(new MethodResponse((int)StatusCode.NotFound));
+                }
+
+                _logger.LogDebug($"Command: component=\"{componentName}\", no temperature readings sent yet, cannot generate any report.");
+                return await Task.FromResult(new MethodResponse((int)StatusCode.NotFound));
+            }
+            catch (JsonReaderException ex)
+            {
+                _logger.LogDebug($"Command input is invalid: {ex.Message}.");
+                return await Task.FromResult(new MethodResponse((int)StatusCode.BadRequest));
+            }
+        }
+
+        private async Task SetDesiredPropertyUpdateCallbackAsync(TwinCollection desiredProperties, object userContext)
+        {
+            bool callbackNotInvoked = true;
+
+            foreach (KeyValuePair<string, object> propertyUpdate in desiredProperties)
+            {
+                string componentName = propertyUpdate.Key;
+                if (_desiredPropertyUpdateCallbacks.ContainsKey(componentName))
+                {
+                    _desiredPropertyUpdateCallbacks[componentName]?.Invoke(desiredProperties, componentName);
+                    callbackNotInvoked = false;
+                }
+            }
+
+            if (callbackNotInvoked)
+            {
+                _logger.LogDebug($"Property: Received a property update that is not implemented by any associated component.");
+            }
+
+            await Task.CompletedTask;
+        }
+
+        // The desired property update callback, which receives the target temperature as a desired property update,
+        // and updates the current temperature value over telemetry and property update.
+        private async Task TargetTemperatureUpdateCallbackAsync(TwinCollection desiredProperties, object userContext)
+        {
+            const string propertyName = "targetTemperature";
+            string componentName = (string)userContext;
+
+            bool targetTempUpdateReceived = PnpHelper.TryGetPropertyFromTwin(desiredProperties,
+                                                                             propertyName,
+                                                                             out double targetTemperature,
```

Comment:
if you are breaking the lines, I suggest you bring the first parameter down to a new line and use only 1 tab.
```
bool targetTempUpdateReceived = PnpHelper.TryGetPropertyFromTwin(
        desiredProperties,
        propertyName,
        out double targetTemperature,
        componentName);
```